completions/bfs.bash

```diff
@@ -0,0 +1,170 @@
+      # when -fprintf is prev1, current word is FILE; perform file completion
+      COMPREPLY=($(compgen -f -- "$cur"))
+      return
+    ;;
+  esac
+
+  # completions with 1-word lookbehind
+  case "$prev" in
+    (-maxdepth|-mindepth|-amin|-Bmin|-cmin|-mmin|-asince|-Bsince|-csince|-msince|-atime|-Btime|-ctime|-mtime|-ilname|-iname|-ipath|-iregex|-iwholename|-inum|-links|-lname|-name|-path|-wholename|-regex|-since|-size|-used|-printf|-newerat|-newerBt|-newerct|-newermt|-xattrname)
+      # arguments whose values should not be completed
+      # (e.g. because they are numeric, glob, regexp, time, etc.)
+      COMPREPLY=()
+      return
+    ;;
+    (-anewer|-Bnewer|-cnewer|-mnewer|-newer|-samefile|-fls|-fprint|-fprint0|-neweraa|-neweraB|-newerac|-neweram|-newerBa|-newerBB|-newerBc|-newerBm|-newerca|-newercB|-newercc|-newercm|-newerma|-newermB|-newermc|-newermm)
+      # arguments whose value is a filename
+      COMPREPLY=($(compgen -f -- "$cur"))
+      return
+    ;;
+    (-f)
+      # arguments whose value is a dirname
+      COMPREPLY=($(compgen -d -- "$cur"))
+      return
+    ;;
+    (-type|-xtype)
+      # -type [bcdlpfswD]
+      #     Find files of the given type
+      # -xtype [bcdlpfswD]
+      #     Find files of the given type, following links when -type would not, and
+      #     vice versa
+      if [[ -n $cur ]] && ! [[ $cur =~ ,$ ]]; then
+        cur+=,
+      fi
+      COMPREPLY=("${cur}b" "${cur}c" "${cur}d" "${cur}l" "${cur}p" "${cur}f" "${cur}s" "${cur}w" "${cur}D")
+      return
+    ;;
+    (-gid|-uid)
+      # -gid [-+]N
+      # -uid [-+]N
+      #     Find files owned by group/user ID N
+      #TODO: list numeric uids/gids
+      COMPREPLY=()
+      return
+    ;;
+    (-group)
+      # -group NAME
+      #     Find files owned by the group NAME
+      COMPREPLY=($(compgen -g -- "$cur"))
+      return
+    ;;
+    (-user)
+      # -user  NAME
+      #     Find files owned by the user NAME
+      COMPREPLY=($(compgen -u -- "$cur"))
+      return
+    ;;
+    (-S)
+      # -S bfs|dfs|ids|eds
+      #     Use breadth-first/depth-first/iterative/exponential deepening search
+      #     (default: -S bfs)
+      COMPREPLY=($(compgen -W 'bfs dfs ids eds' -- "$cur"))
+      return
+    ;;
+    (-D)
+      # -D FLAG
+      #     Turn on a debugging flag (see -D help)
+      COMPREPLY=($(compgen -W 'help cost exec opt rates search stat tree all' -- "$cur"))
+      return
+    ;;
+    (-regextype)
+      # -regextype TYPE
+      #     Use TYPE-flavored regexes (default: posix-basic; see -regextype help)
+      COMPREPLY=($(compgen -W 'help posix-basic posix-extended' -- "$cur"))
+      return
+    ;;
+    (-fstype)
+      # -fstype TYPE
+      #     Find files on file systems with the given TYPE
+      #TODO: parse the mount table for a list of mounted filesystem types
+      COMPREPLY=()
+      return
+    ;;
+    (-perm)
+      # -perm [-]MODE
+      #     Find files with a matching mode
+      if [[ -z "$cur" ]]; then
```

Comment:
I'm not sure how helpful this completion is.

Reply:
Actually maybe it can be helpful to remind the user what the valid syntax is.  But a couple things should probably be changed:

- Numeric modes are octal, so no reason to have 8 or 9 in COMPREPLY
- It would be nice to suggest the symbolic syntax too, u+rw,g+x etc.

Reply:
I added symbolic syntax suggestions, can you please verify that they match the expected syntax?